### PR TITLE
[nexus] add BlueprintZoneDisposition::Expunged, make disposition a column in bp_omicron_zone

### PR DESCRIPTION
**Pull request by sunshowers**
This PR (finally) adds a `BlueprintZoneDisposition::Expunged` enum, along with database support for it by making the disposition a column.

We also drop the `bp_omicron_zones_not_in_service` table -- we don't bother to check for its contents because we aren't ever setting any zones as quiesced on customer sites currently.

I've also included a Diesel extension to allow easily filtering against `BlueprintZoneDisposition`s, along with an example query for it.

### CODE REVIEW

File: schema/crdb/blueprint-disposition-column/up2.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-- Add the disposition column to the bp_omicron_zone table.
+ALTER TABLE omicron.public.bp_omicron_zone
+    ADD COLUMN IF NOT EXISTS disposition omicron.public.bp_zone_disposition
+    NOT NULL
+    -- The only currently-representable zones are in-service and quiesced
+    -- (represented by bp_omicron_zones_not_in_service, which we're going to
+    -- drop in the next statement). We don't actually have any quiesced zones
+    -- yet, so it's fine to just do this.
+    DEFAULT 'in_service';
```
Comment (jgallagher) [inline schema/crdb/blueprint-disposition-column/up2.sql:9]: Do we want another migration step that removes this default?

Comment (sunshowers) [reply schema/crdb/blueprint-disposition-column/up2.sql:9]: Yeah we do I'm pretty sure, thanks for catching!

Comment (sunshowers) [reply schema/crdb/blueprint-disposition-column/up2.sql:9]: Done.

Comment (andrewjstone) [reply schema/crdb/blueprint-disposition-column/up2.sql:9]: Ah, that's clever.

**Review by andrewjstone — APPROVED** [review]
Great stuff @sunshowers

**Pull request merged**